### DISCUSSION TITLE
Use multiple types

### DISCUSSION THREAD

**Discussion by InesaDiamond** [AI-Planning/pddl#121] [Q&A]
In the variables section where I create the domain file, how do you have multiple types? This is roughly what I have: a b c d = variables(“a b c d”, types=[“a”, “b”, “b”, “b”]) But it doesn’t work

**Comment by francescofuggitti** [comment]
This is not the intended use of `variables(...)`. When using `variables(...)`, all the specified types apply to all variables. If I understand correctly, to achieve what you want, you can do the following:

```
a = Variable("a")
b, c, d = variables("b c d", types=["b"])
```
